### PR TITLE
ENH: improve rayleigh accuracy for large x.

### LINKED ISSUE #4034: improve accuracy of scipy.stats.rayleigh distribution for large x
Currently the sf-isf round trip for large x return wrong values as shown here:
In [85]: rayleigh.isf(rayleigh.sf(9,1),1)
Out[85]: 9.0000752648062754

By replacing exp(x)-1 with expm1(x) and log(1+x) with log1p(x) one can improve the accuracy of the rayleigh distribution (_cdf, _sf, and _ppf) to machine precision for large x.

Replacing _cdf  and _ppf with

    def _cdf(self, r):
        return - expm1(-r * r / 2.0)

    def _sf(self, r):
        return exp(-r * r / 2.0)

    def _isf(self, q):
        return sqrt(-2 * log(q))

    def _ppf(self, q):
        return sqrt(-2 * log1p(-q))

one acheives 
In [86]: rayleigh.isf(rayleigh.sf(9,1),1)
Out[86]: 9.0

### PR DESCRIPTION
**Pull request by maniteja123**
Improve the accuracy of the rayleigh distribution for large x, by replacing exp(x)-1 with expm1(x) and log(1+x) with log1p(x). closes #4034

### CODE REVIEW

File: scipy/stats/tests/test_distributions.py [hunk 1]
```diff
@@ -2172,6 +2172,11 @@ def test_lomax_accuracy():
     # regression test for gh-4033
     p = stats.lomax.ppf(stats.lomax.cdf(1e-100,1),1)
     assert_equal(p, 1e-100)
-
+    
+def test_rayleigh_accuracy():
+    # regression test for gh-4034
+    p = stats.rayleigh.isf(stats.rayleigh.sf(9,1),1)
+    assert_equal(p, 9.0)
```
Comment (rgommers) [inline scipy/stats/tests/test_distributions.py]: Exact equality is still not guaranteed; for comparing floating point numbers you should normally use a tolerance. I suggest to replace this by ``assert_almost_equal(p, 9.0, decimal=15)``.

Comment (rgommers) [inline scipy/stats/tests/test_distributions.py]: Besides this line, the PR looks good to me.

**Comment by maniteja123** [comment]
@rgommers , do I need to squash the commits ?

**Comment by rgommers** [comment]
No need to squash these commits.

**Comment by ev-br** [comment]
Somehow this needs a rebase

**Comment by maniteja123** [comment]
@ev-br should I rebase on master ?

**Comment by maniteja123** [comment]
Sorry, I was trying to rebase and it got closed.

**Comment by maniteja123** [comment]
Doing it in #4426 now. It would be helpful if you can tell how to prevent accidentally closing a PR. It has happened before also when I rebased. Please mark this as invalid.

**Comment by rgommers** [comment]
Maybe you somehow pushed a branch with zero commits? The rebase should be as simple as:

    git rebase master    # check that this worked
    git push origin feature-branch -f      # you need -f, or Github will reject your push

where ``feature-branch`` is the name of your branch of course, in this case  ``issue4034``.

**Comment by maniteja123** [comment]
@rgommers . When I did ``git rebase master``, there were some conflicts and it didn't allow further commands until I did one of the options - ``abort ``, ``continue`` or ``skip``. I couldn't understand and then did the ``git push`` as you suggested. I will take advice next time before pushing to the feature branch. Thanks :)

**Pull request closed without merging**